FinSearchUnified/Tests/Components/ConfigLoaderTest.php

```diff
@@ -0,0 +1,425 @@
+                ->method('get')
+                ->willThrowException(new RequestException());
+        } else {
+            $callback = $this->returnCallback(function ($url) use ($httpResponse) {
+                Assert::assertSame($this->expectedUrl, $url);
+
+                return $httpResponse;
+            });
+
+            $httpClientMock->expects($this->once())
+                ->method('get')
+                ->will($this->onConsecutiveCalls($callback));
+        }
+
+        $configLoader = new ConfigLoader(
+            Shopware()->Container()->get('cache'),
+            $httpClientMock,
+            Shopware()->Config()
+        );
+
+        $reflector = new ReflectionObject($configLoader);
+        $method = $reflector->getMethod('getConfigFile');
+        $method->setAccessible(true);
+        $configBody = $method->invoke($configLoader);
+
+        $this->assertSame($expectedResponse, $configBody, $assertionMessage);
+    }
+
+    public function configFileProvider()
+    {
+        return [
+            'Correct url is provided' => [200, 'configBody', 'configBody', 'Expected config url to be correct'],
+            'Url request throws exception' => [200, null, null, 'Expected config to be null'],
+            'Url request is not successful' => [404, 'configBody', null, 'Expected config to be null'],
+            'Url request is successful' => [200, 'configBody', 'configBody', 'Expected config file to be returned'],
+        ];
+    }
+
+    public function testCacheKey()
+    {
+        $configLoader = new ConfigLoader(
+            Shopware()->Container()->get('cache'),
+            Shopware()->Container()->get('http_client'),
+            Shopware()->Config()
+        );
+
+        $expectedCacheKey =
+            sprintf('%s_%s', 'fin_service_config', strtoupper(Shopware()->Config()->offsetGet('ShopKey')));
+
+        $reflector = new ReflectionObject($configLoader);
+        $method = $reflector->getMethod('getCacheKey');
+        $method->setAccessible(true);
+        $key = $method->invoke($configLoader);
+
+        $this->assertSame($expectedCacheKey, $key, 'Incorrect cache key was returned');
+    }
+
+    /**
+     * @dataProvider warmUpCacheProvider
+     *
+     * @param int $responseCode
+     * @param string|null $responseBody
+     * @param array $expectedConfig
+     *
+     * @throws ReflectionException
+     */
+    public function testWarmUpCache($responseCode, $responseBody, array $expectedConfig)
+    {
+        $httpResponse = new Response($responseCode, [], $responseBody);
+        $httpClientMock = $this->createMock(GuzzleHttpClient::class);
+
+        $responseCallback = $this->returnCallback(function ($url) use ($httpResponse) {
+            Assert::assertSame($this->expectedUrl, $url);
+
+            return $httpResponse;
+        });
+
+        $httpClientMock->expects($this->once())
+            ->method('get')
+            ->will($this->onConsecutiveCalls($responseCallback));
```

Comment:
Use `$responseCallback` together with `->with` and use `->will` to return `$httpResponse`.